Neos.Neos/Tests/Behavior/Features/ContentRepository/WorkspaceService.feature

```diff
@@ -87,33 +87,71 @@ Feature: Neos WorkspaceService related features
       | Title               | Description                    | Classification | Owner user id |
       | some-root-workspace | Some new workspace description | ROOT           |               |
 
-
   Scenario: Create a single personal workspace
     When the root workspace "some-root-workspace" is created
     And the personal workspace "some-user-workspace" is created with the target workspace "some-root-workspace" for user "jane.doe"
+    Then the personal workspace for user "jane.doe" is "some-user-workspace"
     Then the workspace "some-user-workspace" should have the following metadata:
       | Title               | Description | Classification | Owner user id |
       | some-user-workspace |             | PERSONAL       | janedoe       |
 
+  Scenario: Personal workspace names are unique https://github.com/neos/neos-development-collection/issues/2850
+    And the following Neos users exist:
+      | Id  | Username  | First name | Last name | Roles                   |
+      | 123 | test-user | Test       | User      | Neos.Neos:Administrator |
+      | 456 | test.user | Test       | User      | Neos.Neos:Administrator |
+
+    When the root workspace "live" is created
+
+    When a personal workspace for user "test-user" is created
+    Then the personal workspace for user "test-user" is "test-user"
+
+    When a personal workspace for user "test.user" is created
+    Then the personal workspace for user "test.user" is "test-user-1"
+
+  Scenario: For multiple personal workspaces only one workspace is returned
+    When the root workspace "some-root-workspace" is created
+    And the personal workspace "b-user-workspace" is created with the target workspace "some-root-workspace" for user "jane.doe"
+    And the personal workspace "a-user-workspace" is created with the target workspace "some-root-workspace" for user "jane.doe"
+    And the personal workspace "c-user-workspace" is created with the target workspace "some-root-workspace" for user "jane.doe"
+    Then the personal workspace for user "jane.doe" is "a-user-workspace"
```

Comment:
maybe for the 9.0 release we should disallow having two personal workspaces per user? Not really via sql index but at least via php api? That would allow us to reduce some insanity like seen here 😅 We can always relax that constraint later.

Reply:
Edit: with https://github.com/neos/neos-development-collection/pull/5334/commits/31473fb4efae71bb62a10665414a6cd545ae4f4b i introduced a constraint to not allow that 😅